src/gui/shareusergroupwidget.cpp

```diff
@@ -926,13 +932,46 @@ void ShareUserLine::customizeStyle()
     _deleteShareButton->setIcon(deleteicon);
 
     _ui->noteConfirmButton->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
-    _ui->confirmExpirationDate->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
     _ui->progressIndicator->setColor(QGuiApplication::palette().color(QPalette::WindowText));
 
     // make sure to force BackgroundRole to QPalette::WindowText for a lable, because it's parent always has a different role set that applies to children unless customized
     _ui->errorLabel->setBackgroundRole(QPalette::WindowText);
 }
 
+QPixmap ShareUserLine::pixmapForShareeType(int type, const QColor &backgroundColor) const
+{
+    const QString pixmapColor = backgroundColor.isValid() && !Theme::isDarkColor(backgroundColor) ? "black" : "white";
+    switch (type) {
+    case Sharee::Room:
+        return QPixmap::fromImage(QImage(QString(":/client/theme/%1/talk-app.svg").arg(pixmapColor)));
+    case Sharee::Email:
+        return QPixmap::fromImage(QImage(QString(":/client/theme/%1/email.svg").arg(pixmapColor)));
+    default:
+        return {};
+    }
+}
+
+QColor ShareUserLine::backgroundColorForShareeType(int type) const
```

Comment:
do not use `int` but `Sharee::Type`
then make sure you mention all enum values as case and remove the default
in case we add another enum value the compiler will help us by emitting a warning

Reply:
@mgallien If I do so, I get a warning that I must use `enum class`. Should I use it then?

Reply:
@mgallien What do you mean `mention all enum values`? Isn't it easier to just process those switch-cases that I am interested in and then just continue with the default logic for everything else? Why should `default` be removed? It's a standard structure for a switch-case to include a default for those cases not processed, has this changed over past years?

Reply:
if you use `default` you loose warnings by compiler when a value is added to the enum but not the `switch`
that is then considered a bad practice and can cause bugs later